drivers/usb/usbuhci_new/usbuhci.h

```diff
@@ -0,0 +1,269 @@
+#ifndef USBUHCI_H__
+#define USBUHCI_H__
+
+#include <ntddk.h>
+#include <windef.h>
+#include <stdio.h>
+#include <hubbusif.h>
+#include <usbbusif.h>
+#include <usbdlib.h>
+#include <drivers/usbport/usbmport.h>
+#include "hardware.h"
+
+#define UHCI_MAX_HC_SCHEDULE_ERRORS        16
+
+#define UHCI_MAX_ISO_TRANSFER_SIZE         0x10000
+#define UHCI_MAX_BULK_TRANSFER_SIZE        0x1000
+#define UHCI_MAX_ISO_TD_COUNT              256
+#define UHCI_MAX_INTERRUPT_TD_COUNT        8
+
+/* Host Controller Driver Transfer Descriptor (HCD TD) */
+#define UHCI_HCD_TD_FLAG_ALLOCATED     0x00000001
+#define UHCI_HCD_TD_FLAG_PROCESSED     0x00000002
+#define UHCI_HCD_TD_FLAG_DONE          0x00000008
+#define UHCI_HCD_TD_FLAG_NOT_ACCESSED  0x00000010
+#define UHCI_HCD_TD_FLAG_DATA_BUFFER   0x00000020
+#define UHCI_HCD_TD_FLAG_GOOD_FRAME    0x00000040
+#define UHCI_HCD_TD_FLAG_CONTROLL      0x00000400
+#define UHCI_HCD_TD_FLAG_STALLED_SETUP 0x00000800
+
+typedef struct _UHCI_ENDPOINT *PUHCI_ENDPOINT;
+typedef struct _UHCI_TRANSFER *PUHCI_TRANSFER;
+
+typedef struct _UHCI_HCD_TD {
+  /* Hardware */
+  UHCI_TD HwTD;
+  /* Software */
+  USB_DEFAULT_PIPE_SETUP_PACKET SetupPacket;
+  ULONG_PTR PhysicalAddress;
```

Comment:
I'm guessing this structure doesn't change its size depending on whether you run a 32 or 64 bit OS. If it's always 32 bits, ULONG is the right choice.

Reply:
Earlier I already got rid of ULONG_PTR, but sent a commit to the "usb_wip" branch. By the way, in usbport there are also several changes. I will work on this after the "Usbport transaction translator" so as not to get confused.